### DISCUSSION TITLE
block paste

### DISCUSSION THREAD

**Discussion by clayton256** [orbitalquark/textadept#177] [General]
I'm sure I'm missing the obvious, but...

After copying a block selection (shift+option arrow) is there a way to paste the block?

Thanks!

**Comment by orbitalquark** [comment]
Yes, make a new rectangular selection where you want to paste over, and then do the paste. Scintilla (Textadept's editing component) does not do "rectangular paste" into an empty selection.

**Reply by clayton256** [reply]
If I'm following you correctly, I tried that as it seems the obvious thing to do. However, I getting different results. For instance, given the text:
`1 = A`
 `2 = B`
 `3 = C`

if I block select the column of numbers and then try to paste the column 2 spaces left, I get the following:
`1 =1`
`2`
`3`
` A`
`2 =1`
`2`
`3`
` B`
`3 =1
`2
`3`
` C`
Haha, What am I doing wrong?

**Reply by rgieseke** [reply]
If i do a multi-select of your example, ctrl-x, then move extended cursor left and paste, it works.

(Edited to remove unrelated setting selection)

**Comment by orbitalquark** [comment]
You need to block select the column of letters and then paste. The column will be replaced with numbers. If you create a zero-width block selection before the letters and paste, the column of numbers will be inserted before the letters. The trick is that you need to paste *into* a block selection, not a single selection.

**Reply by clayton256** [reply]
Yep, that's what I'm doing - cursor on the 1, hold down shift-option and arrow down 2 lines and right 1 column, release shit-option, press command-c, cursor to the right of the =, hold down shift-option and arrow down 2 lines, release shift-option and press command-v.

**Comment by orbitalquark** [comment]
Okay, I'll try this on macOS when I have a chance. I did my tests on Linux.

**Reply by clayton256** [reply]
I just tried this on my linux box. It works as described. So either I've mucked up my install or it's broken on macOS?

**Comment by orbitalquark** [comment]
It looks broken on macOS. It's likely a Scintilla bug that I'll have to dig into and submit an upstream fix for. A workaround is to make the selection and then use the command entry to run:

  replace_rectangular(ui.clipboard_text)

**Comment by orbitalquark** [comment]
Okay, this is silly. It's not a bug in Scintilla, but a shortcoming on macOS. Scintilla has a method of storing rectangular selections on the system clipboard, but macOS does not support this method. I have updated Textadept's manual on this topic[1]. Sorry about that :(

[1]: https://github.com/orbitalquark/textadept/commit/241e28393f9b2aa806dee8d04d8a7f2cd7f6db54

**Comment by clayton256** [comment]
Bummer!  I’ve been using the workaround command replace_rectangular(ui.clipboard_text) but I often type it in wrong so have to start over.

How can I assign the command to the keys command-shift-v? I tried the below but it doesn’t work.
keys['cV'] = function()
    replace_rectangular(ui.clipboard_text)
end

Mark

**Comment by orbitalquark** [comment]
Try `keys['cmd+V'] = function() ... end`. Textadept doesn't use abbreviated modifier keys anymore.

**Comment by clayton256** [comment]
Yes! Thanks, that does it!

Mark
